MH4U Dex/CoreDataController.m

```diff
@@ -517,29 +518,41 @@ - (NSManagedObjectModel *)managedObjectModel
 
 - (NSPersistentStoreCoordinator *)persistentStoreCoordinator
 {
-    // The persistent store coordinator for the application. This implementation creates and return a coordinator, having added the store for the application to it.
     if (_persistentStoreCoordinator != nil) {
         return _persistentStoreCoordinator;
     }
     
-    // Create the coordinator and store
+    NSURL *storeURL = [self.applicationDocumentsDirectory URLByAppendingPathComponent:@"MH4U_DEX.sqlite"];
+    NSFileManager *fileManager = [NSFileManager defaultManager];
     
-    _persistentStoreCoordinator = [[NSPersistentStoreCoordinator alloc] initWithManagedObjectModel:[self managedObjectModel]];
-    NSURL *storeURL = [[self applicationDocumentsDirectory] URLByAppendingPathComponent:@"MH4U_Dex.sqlite"];
-    NSError *error = nil;
-    NSString *failureReason = @"There was an error creating or loading the application's saved data.";
-    if (![_persistentStoreCoordinator addPersistentStoreWithType:NSSQLiteStoreType configuration:nil URL:storeURL options:nil error:&error]) {
-        // TODO: fix all of this later.
-        // Report any error we got.
-        NSDictionary *dict = @{
-                               NSLocalizedDescriptionKey: @"Failed to initialize the application's saved data",
-                               NSLocalizedFailureReasonErrorKey: failureReason,
-                               NSUnderlyingErrorKey: error,
-                               };
-        error = [NSError errorWithDomain:@"YOUR_ERROR_DOMAIN" code:9999 userInfo:dict];
-        // Replace this with code to handle the error appropriately.
-        // abort() causes the application to generate a crash log and terminate. You should not use this function in a shipping application, although it may be useful during development.
-        NSLog(@"Unresolved error %@, %@", error, [error userInfo]);
+    // If the expected store doesn't exist, copy the default store.
+    if (![fileManager fileExistsAtPath:storeURL.path]) {
+        NSURL *defaultStoreURL = [[NSBundle mainBundle] URLForResource:@"PREBUILT_STORE" withExtension:@"sqlite"];
+        if (defaultStoreURL) {
+            NSError *fileCopyError;
+            [fileManager copyItemAtURL:defaultStoreURL
+                                 toURL:storeURL
+                                 error:&fileCopyError];
+            if (fileCopyError) {
```

Comment:
`-[NSFileManager copyItemAtURL:toURL:error:]` returns a `BOOL`, so you should be testing that to test for success.  It's entirely possible that `fileCopyError` will get scribbled on (and be non-`nil`) even when `copyItemAtURL:toURL:error:` succeeds.